src/controller/courses/db/index.js

```diff
@@ -0,0 +1,79 @@
+const {coursesModel} = require('../../../models/courses');
+const {modelUtils} = require('../../../utils/models');
+
+const createCourse = async(req, res) => {
+	try {
+		const course = await coursesModel.create(req.body);
+		res.status(200).send(course);
+	} catch (err) {
+		res.status(400).send({errors: modelUtils.extractErrorMessages(err.errors)});
+	}
+};
+
+const getCourse = async(req, res) => {
+	try {
+		const course = await coursesModel.findByPk(req.params.id);
+
+		if (!course) {
+			res.status(404).send({error: 'No course found with given id'});
+			return;
+		}
+		res.status(200).send(course);
+	} catch (err) {
+		res.status(404).send({errors: modelUtils.extractErrorMessages(err.errors)});
+	}
+};
+
+const getCourses = async(req, res) => {
+	let {perPage, page} = req.query;
+	perPage = (+perPage || 10);
+	page = (+page || 1);
+
+	const total = await coursesModel.count();
+	const courses = await coursesModel.findAll({limit: perPage * page, offset: perPage * (page - 1)});
+	res.status(200).send({
+		total, page, perPage, courses
+	});
+};
+
+const updateCourse = async(req, res) => {
+	try {
+		const [updated] = await coursesModel.update({...req.body}, {
+			where: {id: req.params.id}
+		});
+
+		if (!updated) {
+			res.status(404).send({error: 'No course found with given id'});
```

Comment:
an alternate flow could be:
- first check if the course exists (using `findByPk`)
- if not, throw an error saying `No Course with given ID`
- if yes, try to update it using the `req.body`
    - if success, return the new updated course
    - if failure, a different error saying the body was invalid
    - OR alternatively, we could try to validate the request body even before trying to update

Reply:
Thanks for this comment. I missed this. I have updated the code to reflect the suggested comment